pkg/cli/updater/updater.go

```diff
@@ -188,16 +183,24 @@ func (u *updater) defaultOptions() error {
 		}
 	}
 
-	curChannel, curLocalBuild := u.getVersionInfo(curVersion)
-	if curLocalBuild {
+	curVersion, err := semver.NewVersion(u.version)
+	if err != nil {
 		u.disabled = true
-		u.disabledReason = "using locally built version"
+		u.disabledReason = fmt.Sprintf("failed to parse current version as semver: %v", err)
 	}
 
-	// determine channel from version string if not set
-	if u.channel == "" {
-		u.channel = curChannel
-		u.channelReason = fmt.Sprintf("using %s version", curChannel)
+	if curVersion != nil {
+		curChannel, curLocalBuild := u.getVersionInfo(curVersion)
+		if curLocalBuild {
+			u.disabled = true
```

Comment:
nit: it would be nice to add a test for this since we already have the test suite. Other than that, LGTM.

Reply:
I would, but I don't know a good way to do that in tests currently. I can build a fake out for `debug.ReadBuildInfo`, but that still doesn't help me all that much because I don't know it's behaviour all that well right now. Was hoping to refine as we use it more (it really will only impact go run/install, which is already broke atm)

Reply:
Definitely makes sense.